lib/vsc/utils/ssh_util.py

```diff
@@ -0,0 +1,176 @@
+            logging.error("Authentication to host %s failed, please verify your credentials.", self.host)
+            self.disconnect()
+            return False
+
+        except paramiko.SSHException as sshException:
+            logging.error("Could not establish SSH connection to %s: %s", self.host, sshException)
+            self.disconnect()
+            return False
+
+        except socket.timeout:
+            logging.error("Connection timed out to host: %s", self.host)
+            self.disconnect()
+            return False
+
+        except Exception as ex:
+            logging.error("Problem occured trying to connect to %s error: %s", self.host, ex)
+            self.disconnect()
+            return False
+
+        return True
+
+    def disconnect(self):
+        """disconnect"""
+        logging.debug('Disconnecting to %s', self.host)
+        self.client.close()
+
+    def testconnect(self):
+        """Create a connection."""
+        logging.debug('Testing connection to %s', self.host)
+        if self.connect():
+            logging.debug("Connection seems fine.")
+            self.disconnect()
+            return True
+        else:
+            logging.debug("Could not connect.")
+            self.disconnect()
+            return False
+
+    def run_commands(self, commands):
+        """run a list of commands"""
+        if not isinstance(commands, list):
+            logging.error('commands given is not a list.')
+            raise ValueError('commands given is not a list.')
+
+        if self.connect():
+            output = []
+            for command in commands:
+                logging.debug("Executing command: %s", command)
+
+                if self.agent_forw:
+                    # enable credential forwarding for this session
+                    session = self.client.get_transport().open_session()
+                    paramiko.agent.AgentRequestHandler(session)
+
+                try:
+                    # Don't need stdin
+                    _, stdout, stderr = self.client.exec_command(command, self.comm_timeout)
+                    exit_status = stdout.channel.recv_exit_status()
+
+                except Exception as ex:
+                    logging.error('Could not run command %s. error: %s', command, ex)
+                    exit_status = 1
+
+                try:
+                    ssh_stdout = stdout.read().strip()
+                    ssh_stderr = stderr.read().strip()
+
+                except Exception as ex:
+                    logging.error('Could not read from stdout/stderr. error: %s', ex)
+                    exit_status = 1
+                    ssh_stdout = ""
+                    ssh_stderr = ex
+
+                if exit_status != 0:
+                    logging.error("Command failed: %s. stdout: %s stderr: %s",
+                                  command, ssh_stdout, ssh_stderr)
+                    self.disconnect()
+                    return exit_status, ssh_stderr
```

Comment:
You are returning different things compared to the below. Here you return an integer and an Exception. Below you return an integer and a list (output). This seems odd and error prone.